src/TraceEvent/Computers/StartStopActivityComputer.cs

```diff
@@ -1006,6 +1009,54 @@ bool TryProcessDiagnosticSourceStartEvents(TraceEvent data)
             return false;
         }
 
+        void FixAndProcessAppInsightsEvents(TraceEvent data)
+        {
+            Debug.Assert(data.ProviderGuid == MicrosoftApplicationInsighsDataProvider);
+            Debug.Assert(data.Opcode == TraceEventOpcode.Start || data.Opcode == TraceEventOpcode.Stop);
+
+            if (data.TaskName.Equals("Request", StringComparison.OrdinalIgnoreCase))
+            {
+                if (data.Opcode == TraceEventOpcode.Start)
+                {
+                    // If we have a related activity ID, then this start event
+                    // is already being tracked by an outer activity (e.g. an
+                    // ASP.Net request). Do not process it.
+                    if (!IsTrivialActivityId(data.RelatedActivityID))
+                    {
+                        return;
+                    }
+                    else
+                    {
+                        var extraInfo = (string)data.PayloadByName("Name") ?? (string)data.PayloadByName("Id");
+                        OnStart(data, extraInfo);
+                    }
+                }
+                else
+                {
+                    Debug.Assert(data.Opcode == TraceEventOpcode.Stop);
+                    OnStop(data);
+                }
+            }
+            else if (data.TaskName.Equals("Operation", StringComparison.OrdinalIgnoreCase))
```

Comment:
:memo: Same note here about performance of string comparisons

Reply:
👍 Reviewed the code from AI side. I agree, there is little chance the contract being broken. I'll fix it.

**Details:**
https://raw.githubusercontent.com/Microsoft/ApplicationInsights-dotnet/e8f047f6e48abae0e88a9c77bf65df858c442940/src/Microsoft.ApplicationInsights/Extensibility/Implementation/RichPayloadEventSource.cs